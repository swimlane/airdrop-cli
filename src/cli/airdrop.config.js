module.exports = {
  defaults: {
    package_path: './-/',
    package_root: '/-/',    
  },
  name: 'airdrop'
}