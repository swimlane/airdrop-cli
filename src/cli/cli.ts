import { build, GluegunToolbox } from 'gluegun';

/**
 * Create the cli and kick it off
 */
export async function run(argv?: string[] | string): Promise<GluegunToolbox> {
  // create a CLI runtime
  const cli = build('airdrop')
    .src(__dirname)
    .help()
    .version()
    .exclude([
      'semver',
      'system', 
      'prompt',
      'http',
      'template',
      'patching'
    ])
    .defaultCommand({
      run: async ({ print: { info }, meta: { version }, runtime: { brand } }: GluegunToolbox) => {
        info(`${brand} version ${version()}`)
        info(``)
        info(`  Type ${brand} --help for more info`)
      }
    })
    .command({
      name: 'add-bundle',
      alias: ['ab'],
      run: async (toolbox: GluegunToolbox) => {
        const { parameters, print } = toolbox;
        const force = parameters.options.force ? '--force' : '';
        const optimize = parameters.options.optimize ? '--optimize' : '';

        print.info('Adding packages');
        await cli.run(`a ${parameters.string} ${force}`);

        print.info('Bundling packages');
        await cli.run(`b ${parameters.string} ${force} ${optimize}`);
      }
    })
    .create();

  return cli.run(argv);
}
