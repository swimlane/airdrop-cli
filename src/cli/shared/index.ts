export * from './getMap';
export * from './addPackages';
export * from './bundlePackages';