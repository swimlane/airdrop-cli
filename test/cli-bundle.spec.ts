import { createSandbox, execPkgdrop } from './cli.util';

const TIMEOUT = 100000;

describe('add --bundle', () => {
  let output: string;
  let sandbox: any;

  beforeAll(async () => {
    sandbox = await createSandbox();
    await execPkgdrop(`init -y --offline`);
    output = await execPkgdrop(`add lit-element@2.1.0 --clean --bundle`);
  }, TIMEOUT);

  afterAll(async () => {
    sandbox.clean();
  });

  test('displays console messages', () => {
    expect(output).toMatchSnapshot();
  });

  test('files exist', async () => {
    expect(await sandbox.exists('-/lit-element@2.1.0.bundle.js')).toBe('file');
    expect(await sandbox.exists('-/importmap.json')).toBe('file');
  });

  test('importmap', async () => {
    expect(await sandbox.read('-/importmap.json')).toMatchSnapshot();
  });

  test('can\'t bundle again', async () => {
    const out = await execPkgdrop(`bundle lit-element@2.1.0 --offline`);
    expect(out).toContain('Bundle already exists');
    expect(out).toContain('skipping');
  }, 30000);

  test('can force', async () => {
    const out = await execPkgdrop(`bundle lit-element@2.1.0 --force --offline`);
    expect(out).toMatchSnapshot();
  }, TIMEOUT);
});
