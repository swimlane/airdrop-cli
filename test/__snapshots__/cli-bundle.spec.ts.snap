// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add --bundle can force 1`] = `
"Reading existing importmap
Fetching package info for lit-element@2.1.0
Bundling lit-element@2.1.0
Writing bundle for lit-element@2.1.0
Writing importmap
Done! [XX s]"
`;

exports[`add --bundle displays console messages 1`] = `
"Cleaning output directory
Reading existing importmap
Fetching package information for lit-element@2.1.0
Fetching dependecy tree for lit-element@2.1.0
Extracting tarball for lit-element@2.1.0
Extracting tarball for lit-html@1.1.0
Fetching package info for lit-element@2.1.0
Bundling lit-element@2.1.0
Writing bundle for lit-element@2.1.0
Writing importmap
Done! [XX s]"
`;

exports[`add --bundle importmap 1`] = `
"{
  \\"imports\\": {
    \\"lit-element@2.1.0\\": \\"/-/lit-element@2.1.0.bundle.js\\",
    \\"lit-element@2\\": \\"/-/lit-element@2.1.0.bundle.js\\"
  },
  \\"scopes\\": {}
}"
`;
